File: kotlin/src/main/kotlin/ai/ensoul/sdk/resources/Health.kt
package ai.ensoul.sdk.resources

import ai.ensoul.sdk.EnsoulHttpClient
import io.ktor.client.statement.*
import kotlinx.serialization.json.Json
import kotlinx.serialization.json.JsonObject
import kotlinx.serialization.json.jsonObject

class Health(private val client: EnsoulHttpClient) {

    private val json = Json { ignoreUnknownKeys = true }

    suspend fun check(): JsonObject {
        val response = client.getRaw("/health")
        return json.parseToJsonElement(response.bodyAsText()).jsonObject
    }

    suspend fun ready(): JsonObject {
        val response = client.getRaw("/health/ready")
        return json.parseToJsonElement(response.bodyAsText()).jsonObject
    }

    suspend fun live(): JsonObject {
        val response = client.getRaw("/health/live")
        return json.parseToJsonElement(response.bodyAsText()).jsonObject
    }
}

File: typescript/src/resources/health.ts
import type { HTTPClient } from "../http.js";

export class Health {
  constructor(private readonly client: HTTPClient) {}

  async check(): Promise<Record<string, unknown>> {
    return this.client.getRaw("/health");
  }

  async ready(): Promise<Record<string, unknown>> {
    return this.client.getRaw("/health/ready");
  }

  async live(): Promise<Record<string, unknown>> {
    return this.client.getRaw("/health/live");
  }
}
